### PR TITLE
Add ability to trace a check request

### PR DESCRIPTION
**Pull request by josephschorr**
Provides a trace output when requested, as well as the schema used

Fixes https://github.com/authzed/spicedb/issues/696

Screen Shot 2022-07-19 at 2 48 06 PM
Screen Shot 2022-07-19 at 2 48 09 PM
Screen Shot 2022-07-22 at 6 31 59 PM

### CODE REVIEW

**Comment by josephschorr** [comment]
Waiting on changes in API to be merged and pulled into authzed-go

File: cmd/zed/permission.go [hunk 1]
```diff
@@ -24,6 +32,8 @@ func registerPermissionCmd(rootCmd *cobra.Command) {
 	permissionCmd.AddCommand(checkCmd)
 	checkCmd.Flags().Bool("json", false, "output as JSON")
 	checkCmd.Flags().String("revision", "", "optional revision at which to check")
+	checkCmd.Flags().Bool("trace", false, "requests the debug information of the check from SpiceDB and prints out a trace of the requests")
```
Comment (jzelinskie) [inline cmd/zed/permission.go:35]: I'm not convinced that these flag names make sense. `trace` is so overloaded with distributed tracing and logging.

Maybe something like:
`--resolution-path`
`--resolution-schema`

**Review by jzelinskie — CHANGES_REQUESTED** [review]
The display logic should get teased apart from the CLI flag logic and moved to into the `internal/printers` package.

Other than that, this looks good

File: cmd/zed/permission.go (hunk 1 above)
Comment (josephschorr) [reply cmd/zed/permission.go:35]: Those are verbose to type though; I was going for a short, simple name. Maybe... `--debug`?

**Comment by josephschorr** [comment]
> The display logic should get teased apart from the CLI flag logic and moved to into the internal/printers package.

Done

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**